api/v1beta1/kafkacluster_types.go

```diff
@@ -200,6 +200,8 @@ type BrokerConfig struct {
 	// +kubebuilder:default=120
 	// +optional
 	TerminationGracePeriod *int64 `json:"terminationGracePeriodSeconds,omitempty"`
+	// VirtualHostName for brokers behing envoy ssl
+	VirtualHostName string `json:"virtualHostName,omitempty"`
```

Comment:
there is already a config for this: https://github.com/adobe/koperator/blob/4954280f7fbfe3ad8e04ef341dc7afa3acbd1319/api/v1beta1/kafkacluster_types.go#L419

can we use that instead of adding a new one?

Reply:
I think we can reuse the name (overloading), but we have to add a new field since the current `HostnameOverride` is set at listener level while this new hostname is set for each broker for each TLS listener

Reply:
I agree we need a new field, maybe BrokerHostnameOverride but we can keep it at listener level if we make it a template eg `kafka-a1-az1-b%d.`